earth2studio/models/px/aifs.py

```diff
@@ -0,0 +1,857 @@
+                    dtype=entry["dtype"],
+                ).reshape(entry["shape"])
+
+        # Load interpolation matrix
+        # TODO: Maybe change this to allow for multiple packages?
+        interpolation_package = Package(
+            "https://get.ecmwf.int/repository/earthkit/regrid/db/1/mir_16_linear",
+            cache_options={
+                "cache_storage": Package.default_cache(
+                    "aifs-single-1.0_interpolation_matrix"
+                ),
+                "same_names": True,
+            },
+        )
+        interpolation_matrix_path = interpolation_package.resolve(
+            "9533e90f8433424400ab53c7fafc87ba1a04453093311c0b5bd0b35fedc1fb83.npz"
+        )
+        interpolation_matrix = np.load(interpolation_matrix_path)
+        torch_interpolation_matrix = torch.sparse_csr_tensor(
+            crow_indices=torch.from_numpy(interpolation_matrix["indptr"]),
+            col_indices=torch.from_numpy(interpolation_matrix["indices"]),
+            values=torch.from_numpy(interpolation_matrix["data"]),
+            size=(interpolation_matrix["shape"][0], interpolation_matrix["shape"][1]),
+            dtype=torch.float32,
+        )
+        inverse_interpolation_package = Package(
+            "https://get.ecmwf.int/repository/earthkit/regrid/db/1/mir_16_linear/",
+            cache_options={
+                "cache_storage": Package.default_cache(
+                    "aifs-single-1.0_inverse_interpolation_matrix"
+                ),
+                "same_names": True,
+            },
+        )
+        inverse_interpolation_matrix_path = inverse_interpolation_package.resolve(
+            "7f0be51c7c1f522592c7639e0d3f95bcbff8a044292aa281c1e73b842736d9bf.npz"
+        )
+        inverse_interpolation_matrix = np.load(inverse_interpolation_matrix_path)
+        torch_inverse_interpolation_matrix = torch.sparse_csr_tensor(
+            crow_indices=torch.from_numpy(inverse_interpolation_matrix["indptr"]),
+            col_indices=torch.from_numpy(inverse_interpolation_matrix["indices"]),
+            values=torch.from_numpy(inverse_interpolation_matrix["data"]),
+            size=(
+                inverse_interpolation_matrix["shape"][0],
+                inverse_interpolation_matrix["shape"][1],
+            ),
+            dtype=torch.float32,
+        )
+
+        return cls(
+            model,
+            latitudes=torch.Tensor(supporting_arrays["latitudes"]).reshape(1, 1, -1, 1),
+            longitudes=torch.Tensor(supporting_arrays["longitudes"]).reshape(
+                1, 1, -1, 1
+            ),
+            interpolation_matrix=torch_interpolation_matrix,
+            inverse_interpolation_matrix=torch_inverse_interpolation_matrix,
+            variables=VARIABLES,
+        )
+
+    def get_cos_sin_julian_day(
+        self,
+        time_array: np.datetime64,
+        longitudes: torch.Tensor,
+    ) -> tuple[torch.Tensor, torch.Tensor]:
+        """Get cosine and sine of Julian day"""
```

Comment:
Do you think its work doing some sort of link to the location this was adopted from for future reference?

Reply:
Sadly I could not find this code in anemoi. It ended up being just guess and check to get it right.